modular_skyrat/modules/cyborg/code/robot_upgrades.dm

```diff
@@ -180,23 +181,31 @@
 	name = "engineering advanced materials processor"
 	desc = "allows a cyborg to synthesize and store advanced materials"
 	icon_state = "cyborg_upgrade3"
+	model_type = list(/obj/item/robot_model/engineering)
 	model_flags = BORG_MODEL_ENGINEERING
 
 /obj/item/borg/upgrade/advanced_materials/action(mob/living/silicon/robot/borgo, user)
 	. = ..()
 	if(!.)
 		return
+	if(borgo.hasAdvanced)
+		to_chat(usr, span_warning("This unit already has advanced materials installed!"))
```

Comment:
Don't use usr outside of Topic and Verbs

Reply:
Ah whoops. Fixed